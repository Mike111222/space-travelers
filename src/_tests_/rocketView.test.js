import { render } from '@testing-library/react';
import PRocketView from '../components/rocketView';

describe('tests for rocket components in my profile view', () => {
  test('check snapshot for rockets View in my profile', () => {
    const rocketsData = [
      {
        id: 1, name: 'rocket 1', image: 'image1', description: 'description of rocket1', reserved: false,
      },
      {
        id: 2, name: 'rocket 2', image: 'image2', description: 'description of rocket2', reserved: false,
      },
      {
        id: 3, name: 'rocket 3', image: 'image3', description: 'description of rocket3', reserved: true,
      },
      {
        id: 4, name: 'rocket 4', image: 'image4', description: 'description of rocket4', reserved: true,
      },
    ];

    const reservedData = rocketsData.filter((obj) => obj.reserved === true);

    const { container } = render(<PRocketView reservedData={reservedData} />);
    expect(container).toMatchSnapshot();
  });
});
