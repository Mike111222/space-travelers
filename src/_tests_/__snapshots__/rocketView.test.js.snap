// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for rocket components in my profile view checking snapshot for rockets View in profile 1`] = `
<div>
  <section>
    <div
      class="rockets"
    >
      <h2
        class="sectionTitle"
      >
        Rockets
      </h2>
      <table
        class="holderTable"
      >
        <tbody>
          <tr
            class="rocketrow"
          >
            <td
              class="rocketdata"
            >
              rocket 3
            </td>
          </tr>
          <tr
            class="rocketrow"
          >
            <td
              class="rocketdata"
            >
              rocket 4
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;
